### PR TITLE
Choose element

### LINKED ISSUE #1148: Choose an element
First part done in #1132

- [x] add runner tests
- [x] add e2e test

### PR DESCRIPTION
**Pull request by jperl**
Resolves #1148

### CODE REVIEW

File: runner/src/environment/ElementChooser.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        });
+      });
+
+    await Promise.all(promises);
+  }
+
+  get value(): ElementChooserValue {
+    return {
+      ...(this._lastElementChosen || {}),
+      active: this._active,
```
Comment (jperl) [inline runner/src/environment/ElementChooser.ts:87]: change to isActive

File: app/components/Editor/Sidebar/Snippet/Action.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const width = "156px";
+
+export default function Action({
+  hasText,
+  isFillable,
+  onSelectAction,
+  value,
+}: Props): JSX.Element {
+  const options = buildActionOptions(hasText, isFillable);
+  const optionsHash = options.join("");
```
Comment (flaurida) [inline app/components/Editor/Sidebar/Snippet/Action.tsx:25]: more like a list than a hash 🤔

File: app/components/Editor/Sidebar/Snippet/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  if (action === "Fill test email") {
+    return `const { email, waitForMessage } = getInbox();\nawait page.fill(${selectorArgument}, email);\n// send the email then wait for the message\n// const message = await waitForMessage();`;
+  }
+
+  if (action === "Hover") return `await page.hover(${selectorArgument});`;
+
+  if (action === "Upload image") {
+    return `page.on('filechooser', (chooser) => chooser.setFiles('/root/files/avatar.png'));\nawait page.click(${selectorArgument});`;
```
Comment (flaurida) [inline app/components/Editor/Sidebar/Snippet/helpers.ts:64]: avatar should be default?

File: app/components/Editor/Sidebar/Snippet/index.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        startLine: patchIndex + 1,
+        endLine: patchIndex + 1 + snippetLines.length,
+      },
+      test_id: controller.id,
+      version: controller.version,
+    });
+  };
+
+  useOnHotKey({ hotKey: "Escape", onHotKey: stopElementChooser });
+  useOnHotKey({ hotKey: "Enter", onHotKey: addSnippet, requireMeta: true });
```
Comment (flaurida) [inline app/components/Editor/Sidebar/Snippet/index.tsx:67]: should confirm this does not also run test

File: app/components/Editor/Sidebar/Snippet/Code.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { Box } from "grommet";
```
Comment (jperl) [inline app/components/Editor/Sidebar/Snippet/Code.tsx:1]: Can we share with our docs code component?

File: app/components/shared/AppButton/index.tsx [hunk 6]
```diff
@@ -166,40 +148,51 @@ const StyledAppButton = styled(AppButton)`
...
-      };
+      color: ${textColor[props.hoverType || props.type]};
+      fill: ${textColor[props.hoverType || props.type]};
     }
     `}
 
-    ${(props) => props.type === "dark" && `border-color: ${colors.gray6};`}
+    ${(props) =>
+      !props.isDisabled &&
+      ["dark", "snippet"].includes(props.type) &&
```
Comment (jperl) [inline app/components/shared/AppButton/index.tsx:166]: What is the difference between these two? Can we consolidate dark and snippet type?

**Review by flaurida — APPROVED** [review]

File: app/components/Editor/Sidebar/Snippet/Code.tsx (hunk 5 above)
Comment (flaurida) [reply app/components/Editor/Sidebar/Snippet/Code.tsx:1]: i tried to share what i could. the styles are different for the docs tho

File: app/components/Editor/Sidebar/Snippet/helpers.ts (hunk 3 above)
Comment (jperl) [reply app/components/Editor/Sidebar/Snippet/helpers.ts:64]: I think so

**Pull request merged**